src/connector/src/source/kafka/source/message.rs

```diff
@@ -30,15 +30,15 @@ use crate::source::SourceMeta;
 #[derive(Debug, Clone)]
 pub struct KafkaMeta {
     // timestamp(milliseconds) of message append in mq
-    pub timestamp: Option<i64>,
+    pub timestamp: Timestamp,
     pub headers: Option<OwnedHeaders>,
 }
 
 impl KafkaMeta {
     pub fn extract_timestamp(&self) -> Option<DatumRef<'_>> {
-        self.timestamp.map(|ts| {
+        self.timestamp.to_millis().map(|ts_millis| {
```

Comment:
Similarly
```
        Some(
            risingwave_common::types::Timestamptz::from_millis(self.timestamp.to_millis()?)?.into(),
        )
```

Especially, `unwrap_or_default` can return `1970-01-01T00:00:00Z` when it overflows. It is better to use `null` in this case.